### PR TITLE
Add test for consul enterprise installation

### PR DESCRIPTION
**Pull request by Etiene**
* Adds test for consul enterprise installation
* Fixes `download-url` var check on ubuntu

@infosecgithub

### CODE REVIEW

File: examples/consul-ami/consul.json [hunk 1]
```diff
@@ -51,7 +51,7 @@
   },{
     "type": "shell",
     "inline": [
-      "if [[ -n '{{user `download_url`}}' ]]; then",
+      "if test -n \"{{user `download_url`}}\"; then",
```
Comment (brikis98) [inline examples/consul-ami/consul.json]: Why the change?

File: modules/install-consul/install-consul [hunk 2]
```diff
@@ -334,7 +334,12 @@ function install {
     install_tls_certificates "$path" "$user" "$ca_file_path" "$cert_file_path" "$key_file_path"
   fi
 
-  log_info "Consul install complete!"
+  if command -v consul; then
+    log_info "Consul install complete!";
+  else
+    log_info "Consul was not installed properly.  Aborting.";
```
Comment (brikis98) [inline modules/install-consul/install-consul]: Add "Could not find consul command" to make it clear what about the install failed.

File: test/consul_helpers.go [hunk 3]
```diff
@@ -79,15 +90,20 @@ func runConsulClusterTest(t *testing.T, packerBuildName string, examplesFolder s
...
-	test_structure.RunTestStage(t,"validate", func() {
+	test_structure.RunTestStage(t, "validate", func() {
 		awsRegion := test_structure.LoadString(t, exampleFolder, SAVED_AWS_REGION)
 		terraformOptions := test_structure.LoadTerraformOptions(t, exampleFolder)
+		keyPair := test_structure.LoadEc2KeyPair(t, exampleFolder)
 
-		// Check the Consul servers
-		checkConsulClusterIsWorking(t, CONSUL_CLUSTER_EXAMPLE_OUTPUT_SERVER_ASG_NAME, terraformOptions, awsRegion)
+		if len(enterpriseUrl) > 0 {
+			checkEnterpriseInstall(t, CONSUL_CLUSTER_EXAMPLE_OUTPUT_SERVER_ASG_NAME, terraformOptions, awsRegion, sshUser, keyPair)
```
Comment (brikis98) [inline test/consul_helpers.go]: Shouldn't the enterprise install *also* check the cluster is working?

File: modules/install-consul/install-consul (hunk 2 above)
Comment (brikis98) [reply modules/install-consul/install-consul]: One other thought: check if the `install-consul` script logs the download URL anywhere...

File: test/consul_helpers.go (hunk 3 above)
Comment (Etiene) [reply test/consul_helpers.go]: done

File: examples/consul-ami/consul.json (hunk 1 above)
Comment (brikis98) [reply examples/consul-ami/consul.json]: That... sounds very odd.

BTW, as I mentioned in my other PR, it might be worth allowing both `--download-url` and `--version` to be specified, but if the option is set to an empty string, to just ignore it. That would remove the need for an if-statement here.

File: modules/install-consul/install-consul [hunk 4]
```diff
@@ -205,7 +205,7 @@ function fetch_binary {
     download_url="https://releases.hashicorp.com/consul/${version}/consul_${version}_linux_amd64.zip"
   fi
 
-  log_info "Downloading Consul from $download_url to $DOWNLOAD_PACKAGE_PATH"
+  log_info "Downloading Consul to $DOWNLOAD_PACKAGE_PATH"
```
Comment (brikis98) [inline modules/install-consul/install-consul]: 👍

File: test/consul_enterprise_test.go [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+import (
+	"os"
+	"testing"
+)
+
+// Test the example in the root folder
+func TestConsulInstallFromURLWithUbuntuAmi(t *testing.T) {
+	t.Parallel()
+	runConsulClusterTest(t, "ubuntu16-ami", ".", "../examples/consul-ami/consul.json", "ubuntu", os.Getenv("CONSUL_AMI_TEMPLATE_VAR_DOWNLOAD_URL"))
```
Comment (brikis98) [inline test/consul_enterprise_test.go]: See my comment in the Vault PR about adding a separate method that reports a clear error if this env var is not set and has a comment explaining why we use an env var.

File: examples/consul-ami/consul.json (hunk 1 above)
Comment (Etiene) [reply examples/consul-ami/consul.json]: Oh that was the original plan in fact, but after discussing with Josh we came to the conclusion that this makes it clearear to users that these two settings are independent and one of them is required

Comment (brikis98) [reply examples/consul-ami/consul.json]: Right, I still think that would be the case. If you specify:

```
--download-url "foo" --version "bar"
```

You'd get an error. But if you specify:

```
--download-url "foo" --version ""
```

That is, an empty string for one of the params, it's ignored, and the other is used. 

And if you specify:

```
--download-url "" --version ""
```

The defaults are used.

Comment (Etiene) [reply examples/consul-ami/consul.json]: But the defaults are specified in the packer template, not in the install script.

Comment (Etiene) [reply examples/consul-ami/consul.json]: You can already pass both of them to the script, it is not clear from the template which one takes precedence. This is why this if is there. To make it clear.
